### PR TITLE
Material properties - acikek's part

### PR DESCRIPTION
**Pull request by acikek**
## Additions

- `level` supports farmland
- `switched` supports hoppers
- `type` supports cave vines

### CODE REVIEW

File: plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialBlockType.java [hunk 1]
```diff
@@ -99,9 +102,13 @@ public Campfire getCampfire() {
         return (Campfire) material.getModernData();
     }
 
-    /*public PointedDripstone getDripstone() { // TODO: 1.17
+    public PointedDripstone getDripstone() {
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialBlockType.java]: Did you think about what you were doing or did you just intentionally break this

File: plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialBlockType.java [hunk 2]
```diff
@@ -151,9 +162,12 @@ else if (isCampfire()) {
             else if (isPistonHead() && mechanism.requireEnum(false, TechnicalPiston.Type.values())) {
                 getPistonHead().setType(TechnicalPiston.Type.valueOf(mechanism.getValue().asString().toUpperCase()));
             }
-            else {
+            else if (isDripstone()){
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialBlockType.java]: formatted wrong

File: plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialBlockType.java [hunk 3]
```diff
@@ -151,9 +162,12 @@ else if (isCampfire()) {
...
                 MultiVersionHelper1_17.materialBlockTypeRunMech(mechanism, this);
             }
+            else if (isCaveVines()) {
+                getCaveVines().setBerries(CoreUtilities.equalsIgnoreCase(mechanism.getValue().asString(), "berries"));
+            }
         }
     }
-}
+}
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialBlockType.java]: broke the formatting here, NL@EoF

File: plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialLevel.java [hunk 4]
```diff
@@ -199,4 +219,4 @@ public void adjust(Mechanism mechanism) {
             setCurrent(level);
         }
     }
-}
+}
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialLevel.java]: NL@EoF

File: plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialSwitchable.java [hunk 5]
```diff
@@ -210,4 +223,4 @@ public void adjust(Mechanism mechanism) {
             setState(mechanism.getValue().asBoolean());
         }
     }
-}
+}
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialSwitchable.java]: NL@EoF

File: plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialBlockType.java (hunk 1 above)
Comment (acikek) [reply plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialBlockType.java]: That was Aya, I just didn't see this when I was porting over changes

File: plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialBlockType.java [hunk 6]
```diff
@@ -103,6 +106,10 @@ public Campfire getCampfire() {
         return (PointedDripstone) material.getModernData();
     }*/
 
+    public CaveVinesPlant getCaveVines() {
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialBlockType.java]: needs the same special handling that dripstone has

**Pull request merged**